oap-storage/oap-storage-cloud/src/main/java/oap/storage/cloud/FileSystem.java

```diff
@@ -259,6 +262,19 @@ public void copy( CloudURI source, CloudURI destination, Map<String, String> use
         }
     }
 
+    public PageSet<? extends StorageMetadata> list( String path, String container, ListContainerOptions options ) {
```

Comment:
I suggest not exposing the jclouds API. For the cloud list options, it makes sense to create our class.